### PR TITLE
[Gabriel Yang] iP

### PR DESCRIPTION
**Pull request by gabyang**
## DukePro

>"Forgive my manners, call me the Duke. Now to business. Weapons, ammunition, healing salves – anything you desire, I can provide." - [The Duke Resident Evil](https://residentevil.fandom.com/wiki/The_Duke/quotes)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is:

1. download it [here](https://github.com/nus-cs2103-AY2223S1)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
``

### CODE REVIEW

File: src/main/java/command/ByeCommand.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import task.DukeTask;
+
+import java.util.ArrayList;
+
+/**
+ * Exit Command
+ */
+public class ByeCommand extends Command {
+    /**
+     * Does Not do Anything
```
Comment (HoJunHao2000) [inline src/main/java/command/ByeCommand.java:17]: - Do leave a line separating the description and parameters. 
- The code does something not nothing. 
- Do add a description for each parameter too! 
- All sentences should end with .
``` java
/**
 * Prints the exit message.
 *      
 * @param taskList
 * @param ui
 * @param storage
 * @throws InvalidFormatException
 */
```

File: src/main/java/command/DeadlineCommand.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    /**
+     * Deconstruct a deadline command based on cmd
+     * @param tasklist
+     * @param ui
+     * @param storage
+     * @throws InvalidFormatException
+     */
+    @Override
+    public void deconstruct(ArrayList<DukeTask> tasklist, Ui ui, Storage storage) throws InvalidFormatException {
```
Comment (HoJunHao2000) [inline src/main/java/command/DeadlineCommand.java]: - Do use OOP principles and separate components of Ui, Parser, and Commands apart from each other
- Consider moving all print statements into the Ui class. You have access to it through the ui argument in deconstruct
- Parsing should be done before passing into the Command object subclasses

File: src/main/java/command/DeadlineCommand.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Represents a deadline command
+ */
+public class DeadlineCommand extends Command {
+    private String cmd;
```
Comment (HoJunHao2000) [inline src/main/java/command/DeadlineCommand.java:19]: Try to use meaning full variable names instead. E.g. command. Should only use shortened names if names are too long or for temporary variables (e.g. in for loops)

File: src/main/java/duke/Duke.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        tasklist = new ArrayList<DukeTask>();
+        Storage.setOnce(tasklist, filePath);
+    }
+
+    /**
+     * Run the program
+     */
+    public void run() {
+        storage.read();
+        Scanner input = new Scanner(System.in);
```
Comment (HoJunHao2000) [inline src/main/java/duke/Duke.java]: you can consider moving your scanner into ui

File: src/main/java/duke/Parser.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import command.*;
```
Comment (HoJunHao2000) [inline src/main/java/duke/Parser.java:3]: Do not use wildcard imports. List all imports down individually.

File: src/main/java/duke/Storage.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * A class that writes and read from files
+ */
+public class Storage {
+
+    private static String FILEPATH;
+    private static ArrayList<DukeTask> tasklist;
+    
+    Storage() {}
```
Comment (HoJunHao2000) [inline src/main/java/duke/Storage.java:19]: Unneccessary

File: src/main/java/task/DukeTask.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Represent a generic task
+ */
+public class DukeTask {
+    private String task;
+    private boolean isMarked;
+    private char taskType;
+
+    public DukeTask () {
```
Comment (HoJunHao2000) [inline src/main/java/task/DukeTask.java:11]: Missing javadocs for many public methods

File: src/main/java/task/DukeTaskDeadline.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+package task;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represent a deadline task
+ */
+public class DukeTaskDeadline extends DukeTask {
+    
+    private LocalDateTime ldt;
```
Comment (HoJunHao2000) [inline src/main/java/task/DukeTaskDeadline.java:11]: use meaningful names

**Review by HoJunHao2000 — COMMENTED** [review]
- Alot of works need to be done on the javadocs
- try to use meaningful names for the variables
- use OOP concepts

**Comment by ThomasHoooo** [comment]
Hey awesome application! Can't wait to try out your product!

File: src/main/java/command/ByeCommand.java (hunk 1 above)
Comment (gabyang) [reply src/main/java/command/ByeCommand.java:17]: Got it thanks!

File: src/main/java/command/DeadlineCommand.java (hunk 2 above)
Comment (gabyang) [reply src/main/java/command/DeadlineCommand.java]: Ok definitely!

File: src/main/java/command/DeadlineCommand.java (hunk 3 above)
Comment (gabyang) [reply src/main/java/command/DeadlineCommand.java:19]: I'll make note of that!

File: src/main/java/duke/Duke.java (hunk 4 above)
Comment (gabyang) [reply src/main/java/duke/Duke.java]: Yep

File: src/main/java/duke/Parser.java (hunk 5 above)
Comment (gabyang) [reply src/main/java/duke/Parser.java:3]: Ok!

**Pull request closed without merging**